system/lib/pthread/library_pthread.c

```diff
@@ -355,8 +356,8 @@ static void _do_call(em_queued_call* q) {
 typedef struct CallQueue {
   void* target_thread;
   em_queued_call** call_queue;
-  int call_queue_head; // Shared data synchronized by call_queue_lock.
-  int call_queue_tail;
+  atomic_int call_queue_head; // Shared data synchronized by call_queue_lock.
+  atomic_int call_queue_tail;
```

Comment:
It looks like these don't need to be atomic (or atomically accessed) because they are already protected by the `call_queue_lock`. It is redundant to use both atomics and a mutex. Am I missing something?

Reply:
I didn't actually look at how the code works.. this was just a blind conversion so it should be NFC.   What do you think about landing this and leaving any other improvements for later?